lib/Sema/CSSimplify.cpp

```diff
@@ -7802,6 +7802,12 @@ ConstraintSystem::simplifyKeyPathApplicationConstraint(
     return SolutionKind::Unsolved;
   };
 
+  // When locator points to a KeyPathDynamicMemberLookup, skip the
+  // key path application.
+  if (locator.getBaseLocator()->isForKeyPathDynamicMemberLookup()) {
```

Comment:
Ah sorry, I just noticed you're using `getBaseLocator` here, which may not include elements added to the locator via the locator builder. You can instead call `locator.last()` and see if the element `isKeyPathDynamicMember()`.

Reply:
+1, I think we might even want to avoid exposing `getBaseLocator` or outright remove it....

Reply:
Makes sense, just fixed :)